frontend/src/views/admin/assignments/offer-actions.js

```diff
@@ -59,49 +61,40 @@ function OfferActionButtons(props) {
         setOfferForAssignmentRejected,
     } = props;
 
-    const [
-        showWithdrawConfirmation,
-        setShowWithdrawConfirmation,
-    ] = React.useState(false);
-
     function createOffers() {
-        for (const assignment of selectedAssignments) {
-            offerForAssignmentCreate(assignment);
-        }
-    }
-    function confirmOfferWithdraw() {
-        // if withdrawing multiple offers at once, show confirmation
-        if (selectedAssignments?.length > 1) {
-            setShowWithdrawConfirmation(true);
-        } else {
-            // does not need confirmation if only withdrawing one offer
-            withdrawOffers();
-        }
+        return selectedAssignments.map((assignment) =>
+            offerForAssignmentCreate(assignment)
+        );
     }
+
     function withdrawOffers() {
-        for (const assignment of selectedAssignments) {
-            offerForAssignmentWithdraw(assignment);
-        }
+        return selectedAssignments.map((assignment) =>
```

Comment:
You need to return a `Promise.all`. Otherwise `await withdrawOffers()` won't do anything.